src/Oauth2Client.php

```diff
@@ -0,0 +1,320 @@
+<?php
+/**
+ * Shapeways API Oauth2 Client
+ * @copyright 2018 Shapeways <api@shapeways.com> (http://developers.shapeways.com)
+ */
+
+namespace Shapeways;
+
+/**
+ * Exception raised when method parameters are not valid
+ */
+class ParameterValidationException extends \Exception
+{
+}
+
+/**
+ * API Oauth2Client for obtaining OAuth2 access token and making
+ * API calls to api.shapeways.com
+ */
+class Oauth2Client
+{
+
+  /**
+   * This determines where authentication requests will be sent and received by your app.
+   * @var string redirectUrl the oauth2 callback url
+   */
+  public $redirectUrl;
+
+  /**
+   * @var string $clientId
+   * @var string $clientSecret
+   */
+  public $clientId, $clientSecret;
+
+  /**
+   * @var string $accessToken the oauth token used for requests
+   * @var string $refreshSecret the oauth secret used for getting new access token
+   */
+  public $accessToken, $refreshToken;
+
+  /**
+   * @var string $baseUrl the api base url used to generate api urls
+   */
+  public $baseApiUrl = 'https://api.shapeways.com';
+
+  /**
+   * @var string $apiVersion the api version used to generate api urls
+   */
+  public $apiVersion = 'v1';
+
+  /**
+   * Create a new \Shapeways\Oauth2CurlClient
+   *
+   * @param string $clientId your app consumer key
+   * @param string $clientSecret your app consumer secret
+   * @param string|null $redirectUrl your app callback url
+   * @param string|null $accessToken a users oauth token if it is already known
+   * @param string|null $refreshToken if it is already known
+   */
+  public function __construct(
+    $clientId,
+    $clientSecret,
+    $redirectUrl,
+    $accessToken = null,
+    $refreshToken = null
+  ) {
+    $this->clientId = $clientId;
```

Comment:
if the client id, secret, and redirect url are all set at the time of construct, why make them public and changeable afterwards?